### PR TITLE
refactor: Easier computation for track params from seed

### PR DESCRIPTION
**Pull request by CarloVarni**
Should give the same results but ~some% faster since there are less computations happening

### CODE REVIEW

**Comment by CarloVarni** [comment]
@paulgessinger maybe I'm missing something obvious... but I see no reason to go to the uv plane for this computation. Why that complication?

File: Core/include/Acts/Seeding/EstimateTrackParamsFromSeed.hpp [hunk 1]
```diff
@@ -214,24 +214,33 @@ std::optional<BoundVector> estimateTrackParamsFromSeed(
...
-  Vector2 uv1 = uvTransform(local1);
-  Vector2 uv2 = uvTransform(local2);
+  // In the new frame the bottom sp is at the origin, while the middle
+  // sp in along the x axis. As such, the x-coordinate of the circle is
+  // at: x-middle / 2.
+  // The y coordinate can be found by using the straight line passing
+  // between the mid point between the middle and top sp and tangent to
+  // the line connecting them
+  Vector2 circleCenter;
+  circleCenter(0, 0) = 0.5 * local1(0, 0);
```
Comment (paulgessinger) [inline Core/include/Acts/Seeding/EstimateTrackParamsFromSeed.hpp:224]: Any reason for the 2D access syntax for a 1D vector?

File: Core/include/Acts/Seeding/EstimateTrackParamsFromSeed.hpp [hunk 2]
```diff
@@ -214,24 +214,33 @@ std::optional<BoundVector> estimateTrackParamsFromSeed(
...
+
+  ActsScalar DeltaX21 = local2(0, 0) - local1(0, 0);
+  ActsScalar SumX21 = local2(0, 0) + local1(0, 0);
```
Comment (paulgessinger) [inline Core/include/Acts/Seeding/EstimateTrackParamsFromSeed.hpp:227]:
```suggestion
  ActsScalar deltaX21 = local2(0, 0) - local1(0, 0);
  ActsScalar sumX21 = local2(0, 0) + local1(0, 0);
```

**Review by paulgessinger — COMMENTED** [review]
From what I understand about your calculation, I think it should work. Do we have a  unit test for this? If not, it should be fairly easy to write (just use three random points) and check that it gives the same radius?

**Comment by CarloVarni** [comment]
@paulgessinger there are tests indeed. My first attempt was indeed breaking some unit tests due to a sign error in my computation: see https://github.com/acts-project/acts/blob/main/Tests/UnitTests/Core/Seeding/EstimateTrackParamsFromSeedTest.cpp

File: Core/include/Acts/Seeding/EstimateTrackParamsFromSeed.hpp (hunk 1 above)
Comment (CarloVarni) [reply Core/include/Acts/Seeding/EstimateTrackParamsFromSeed.hpp:224]: my ignorance on how Eigen works ... no idea I could use only one number ...

**Comment by CarloVarni** [comment]
I believe the failing tests are due to double-precision level differences in the parameters resulting from the different computation. For instance:
```console
Params (0, 0) : -0.621587 vs -0.621587 diff=0 -> -0 %
Params (1, 0) : -16.0916 vs -16.0916 diff=0 -> -0 %
Params (2, 0) : -1.71712 vs -1.71712 diff=0 -> -0 %
Params (3, 0) : 2.97129 vs 2.97129 diff=0 -> 0 %
Params (4, 0) : -0.0284886 vs -0.0284886 diff=1.97758e-16 -> -6.94167e-15 %
Params (5, 0) : 496.005 vs 496.005 diff=5.68434e-14 -> 1.14602e-16 %
```

**Comment by CarloVarni** [comment]
and here is the timing comparison between the nominal (blue) and the modified (red) computation
Screenshot 2023-11-20 alle 8 38 13 PM

**Comment by CarloVarni** [comment]
@paulgessinger are we good with these changes?

**Review by andiwand — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [6a028d84883a53a7048f3d0e88d3c6ee495b2bf0]
### :white_check_mark: All tests successful

|status|job|
|------|---|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34159049) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsBenchmarkWithSpot.sh 8 100](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34159048) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34159047) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateAmbiguityResolution.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34159046) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateResolvedTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34159045) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34159044) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34159043) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34159042) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34159041) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34159040) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34159039) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34159038) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsGSFRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34159037) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34159036) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34159035) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34159034) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34159033) |
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34159032) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34159031) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34159030) |

**Pull request merged**